include/highfive/h5easy_bits/H5Easy_public.hpp

```diff
@@ -0,0 +1,221 @@
+/*
+ *  Copyright (c), 2017, Adrien Devresse <adrien.devresse@epfl.ch>
+ *
+ *  Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ *
+ */
+#ifndef H5EASY_BITS_PUBLIC_HPP
+#define H5EASY_BITS_PUBLIC_HPP
+
+#include "../H5Easy.hpp"
+
+namespace H5Easy {
+
+inline CompressionLevel::CompressionLevel(unsigned deflate_level) : m_deflate_level(deflate_level)
+{
+}
+
+inline unsigned CompressionLevel::get() const
+{
+    return m_deflate_level;
+}
+
+inline void DumpOptions::set(DumpMode mode)
+{
+    if (mode == DumpMode::Create) {
+        m_overwrite = false;
+    }
+    else if (mode == DumpMode::Overwrite) {
+        m_overwrite = true;
+    }
+    else {
+        throw std::runtime_error("Unknown DumpMode");
+    }
+}
+
+inline void DumpOptions::set(Flush mode)
+{
+    if (mode == Flush::False) {
+        m_flush = false;
+    }
+    else if (mode == Flush::True) {
+        m_flush = true;
+    }
+    else {
+        throw std::runtime_error("Unknown Flush");
+    }
+}
+
+inline void DumpOptions::set(Compression level)
+{
+    if (level == Compression::False) {
+        m_compress = false;
+    }
+    else if (level == Compression::Medium || level == Compression::High) {
+        m_compress = true;
+        m_deflate_level = static_cast<unsigned>(level);
+    }
+    else {
+        throw std::runtime_error("Unknown Compression");
+    }
+}
+
+inline void DumpOptions::set(const CompressionLevel& level)
+{
+    m_compress = true;
+    m_deflate_level = level.get();
+}
+
+template <class T, class... Args>
+inline void DumpOptions::set(T arg, Args... args)
+{
+    set(arg);
+    set(args...);
+}
+
+inline void DumpOptions::setDeflateLevel(unsigned level)
+{
+    m_deflate_level = level;
+}
+
+template <class T>
+inline void DumpOptions::setChunkSize(const std::vector<T>& shape)
+{
+    m_chunk_size = std::vector<hsize_t>(shape.begin(), shape.end());
+}
+
+inline void DumpOptions::setChunkSize(std::initializer_list<size_t> shape)
+{
+    m_chunk_size = std::vector<hsize_t>(shape.begin(), shape.end());
+}
+
+inline bool DumpOptions::isOverwrite() const
+{
+    return m_overwrite;
+}
+
+inline bool DumpOptions::isFlush() const
+{
+    return m_flush;
+}
+
+inline bool DumpOptions::isCompress() const
+{
+    return m_compress;
+}
+
+inline unsigned DumpOptions::getDeflateLevel() const
+{
+    return m_deflate_level;
+}
+
+inline bool DumpOptions::isChunkSize() const
```

Comment:
```suggestion
inline bool DumpOptions::isChunked() const
```
? What exactly is this supposed to do?

Reply:
To distinct if the chunk-size if prescribed by the user, or if the dump routines should proceed to automatically choose it.

Reply:
In terms of naming, the `ChunkSize` can now also be used when writing extendable DataSets (to overwrite the default `10` along all dimensions). These DataSets are by definition chunked.

Reply:
In the end I think that `isChunked` is at least as good. So I changed it.